test/experimental/test_vectors.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+import torch
+
+from test.common.assets import get_asset_path
+from test.common.torchtext_test_case import TorchtextTestCase
+from torchtext.experimental.vectors import (
+    Vectors,
+    vectors_from_csv_file
+)
+
+
+class TestVectors(TorchtextTestCase):
+
+    def test_has_unk(self):
+        tensorA = torch.Tensor([1, 0])
+        expected_unk_tensor = torch.Tensor([0, 0])
+
+        tokens = ['a']
+        vectors = [tensorA]
+        vectorsObj = Vectors(tokens, vectors)
+
+        self.assertEqual(vectorsObj['not_in_it'].tolist(), expected_unk_tensor.tolist())
+
+    def test_vectors_basic(self):
+        tensorA = torch.Tensor([1, 0])
+        tensorB = torch.Tensor([0, 1])
+        unk_tensor = torch.Tensor([0, 0])
+
+        tokens = ['a', 'b']
+        vectors = [tensorA, tensorB]
+        vectorsObj = Vectors(tokens, vectors, unk_tensor=unk_tensor)
+
+        self.assertEqual(vectorsObj['a'].tolist(), tensorA.tolist())
+        self.assertEqual(vectorsObj['b'].tolist(), tensorB.tolist())
+        self.assertEqual(vectorsObj['not_in_it'].tolist(), unk_tensor.tolist())
+
+    def test_mismatch_vectors_tokens(self):
+        tensorA = torch.Tensor([1, 0, 0])
+        tensorB = torch.Tensor([0, 1, 0])
+        expected_unk_tensor = torch.Tensor([0, 0, 0])
+
+        tokens = ['a', 'b', 'c']
+        vectors = [tensorA, tensorB]
+        vectorsObj = Vectors(tokens, vectors)
+
+        self.assertEqual(vectorsObj['a'].tolist(), tensorA.tolist())
+        self.assertEqual(vectorsObj['b'].tolist(), tensorB.tolist())
+        self.assertEqual(vectorsObj['c'].tolist(), expected_unk_tensor.tolist())
+
+    def test_vectors_from_file(self):
+        asset_name = 'vectors_test.csv'
+        asset_path = get_asset_path(asset_name)
+        f = open(asset_path, 'r')
+        vectorsObj = vectors_from_csv_file(f)
+
+        expected_tensorA = torch.Tensor([1, 0, 0])
```

Comment:
Do we support both long() and float() tensor for the vector class?

Reply:
hmm let me convert everything to float for now

Reply:
I think it's fine to cover both (actually we should not put any assumptions here). Just make sure you have covered one or two cases for each category.

Reply:
Isn't it important to stick to one type for now since the factory functions will create a `float` type tensor (that's what I am doing now at least). Or do we want to pass in an optional argument to the factory function that specifies the type?

Reply:
@zhangguanheng66 lmk what you think?